### PR TITLE
Password Reset

### PR DESCRIPTION
**Pull request by benvogler**
This PR includes:
- Full password reset flow
- Updated meta tags so that the site's title/description aren't React App 👍 
- Updated favicon so only the inside of the circle is white, for readability on dark themes
- Increased favicon size

### CODE REVIEW

File: client/src/components/BeginPasswordReset/BeginPasswordReset.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if (event.preventDefault) event.preventDefault();
+
+        if (name === 'email') {
+            errors.email = validEmailRegex.test(value)
+                ? ''
+                : 'Email is not valid';
+        }
+
+        this.setState({errors, [name]: value})
+    }
```
Comment (njd9419) [inline client/src/components/BeginPasswordReset/BeginPasswordReset.js]: Any reason to have this abstracted out beyond validating emails?

File: client/src/components/FinishPasswordReset/FinishPasswordReset.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                        <span>
+                            Reset Your Password
+                        </span>
+                    </Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    {!success &&
+                        <Form className="password-reset-form" noValidate onSubmit={event => this.handleSubmit(event)}>
+                            <Form.Group>
+                                <Form.Label>Password</Form.Label>
```
Comment (njd9419) [inline client/src/components/FinishPasswordReset/FinishPasswordReset.js]: What's this field for?

File: client/src/pages/BeginPasswordResetPage/BeginPasswordResetPage.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import BeginPasswordReset from '../../components/BeginPasswordReset/BeginPasswordReset';
+import { beginPasswordReset } from '../../actions/authentication';
+import Spinner from '../../components/Spinner/Spinner';
+
+class BeginPasswordResetPage extends React.Component {
+
+    static mapStateToProps(state) {
+        console.log(state);
```
Comment (njd9419) [inline client/src/pages/BeginPasswordResetPage/BeginPasswordResetPage.js]: Minor: Debug print.

File: src/main/java/com/monumental/services/UserService.java [hunk 4]
```diff
@@ -58,23 +59,41 @@ public User signup(CreateUserRequest userRequest, String appUrl, Locale locale)
...
+        }
+        this.sendPasswordResetEmail(user, this.generateVerificationToken(user, VerificationToken.Type.PASSWORD_RESET), appUrl, locale);
+    }
+
+    public void deleteAllVerificationTokensOfTypeForUser(User user, VerificationToken.Type type) {
+        List<VerificationToken> tokens = this.tokenRepository.findAllByUserAndType(user, type);
+        if (tokens != null && tokens.size() > 0) {
+            this.tokenRepository.deleteAll(tokens);
+        }
+    }
```
Comment (njd9419) [inline src/main/java/com/monumental/services/UserService.java]: Couldn't we just use something like `this.tokenRepository.deleteAllByUserAndType(user, type);`?

File: src/main/resources/messages/email.properties [hunk 5]
```diff
@@ -1 +1,3 @@
-registration.success=Success! You have registered. Please click the link below to verify your email address. 
\ No newline at end of file
+registration.success=Success! You have registered. Please click the link below to verify your email address.
+password-reset.begin=Please click the link below to finish resetting your password.
+password-reset.success=You have successfully reset your Monuments and Memorials password.
```
Comment (njd9419) [inline src/main/resources/messages/email.properties]: Should we make the email subjects messages too?

File: src/main/java/com/monumental/services/UserService.java [hunk 6]
```diff
@@ -58,23 +59,41 @@ public User signup(CreateUserRequest userRequest, String appUrl, Locale locale)
...
+
+    public VerificationToken generateVerificationToken(User user, VerificationToken.Type type) {
+        this.deleteAllVerificationTokensOfTypeForUser(user, type);
```
Comment (njd9419) [inline src/main/java/com/monumental/services/UserService.java]: Then you could just remove the function above and replace it with that call I mentioned.

File: client/src/components/BeginPasswordReset/BeginPasswordReset.js (hunk 1 above)
Comment (benvogler) [reply client/src/components/BeginPasswordReset/BeginPasswordReset.js]: Nope, this was copy and pasted from somewhere else with multiple fields. Fixed now

File: client/src/components/FinishPasswordReset/FinishPasswordReset.js (hunk 2 above)
Comment (benvogler) [reply client/src/components/FinishPasswordReset/FinishPasswordReset.js]: Deleted

File: src/main/resources/messages/email.properties (hunk 5 above)
Comment (benvogler) [reply src/main/resources/messages/email.properties]: We're going to have to redo this all to use HTML emails probably so I haven't bothered

File: src/main/java/com/monumental/repositories/VerificationTokenRepository.java [hunk 7]
```diff
@@ -10,6 +10,6 @@
 
     VerificationToken getByToken(String token);
 
-    List<VerificationToken> findAllByUser(User user);
+    List<VerificationToken> deleteAllByUserAndType(User user, VerificationToken.Type type);
```
Comment (njd9419) [inline src/main/java/com/monumental/repositories/VerificationTokenRepository.java]: Idk if we need to have this return anything.

Comment (benvogler) [reply src/main/java/com/monumental/repositories/VerificationTokenRepository.java]: Fixed

**Pull request merged**